lib/cpp/CMakeLists.txt

```diff
@@ -20,6 +20,8 @@
 # Remove the following once lib/cpp no longer depends on boost headers:
 include(BoostMacros)
 REQUIRE_BOOST_HEADERS()
+set(BOOST_COMPONENTS locale)
+REQUIRE_BOOST_LIBRARIES(BOOST_COMPONENTS)
```

Comment:
I thought we removed all runtime library dependencies on boost linked libraries?

Reply:
no,thrift/protocol/TJSONProtocol.cpp uses boost::locale,not to mention the boost/noncopyable.hpp

Reply:
By repushing this PR,now it would not change the boost code.